modules/library/src/test/scala/zio/elasticsearch/FunctionScoreFunctionsSpec.scala

```diff
@@ -0,0 +1,361 @@
+package zio.elasticsearch
+
+import zio.elasticsearch.ElasticQuery.matches
+import zio.elasticsearch.query.DecayFunctionType._
+import zio.elasticsearch.query.FieldValueFactorFunctionModifier.LOG
+import zio.elasticsearch.query.FunctionScoreFunction._
+import zio.elasticsearch.query.MultiValueMode.Max
+import zio.elasticsearch.query._
+import zio.elasticsearch.script.Script
+import zio.elasticsearch.utils.RichString
+import zio.test.Assertion._
+import zio.test._
+
+object FunctionScoreFunctionsSpec extends ZIOSpecDefault {
+
+  def spec: Spec[TestEnvironment, Any] =
+    suite("functionScore")(
```

Comment:
Does it make sense to put this in ElasticQuery Spec? Instead of having a separate file?